tests/ui/test_ui_tools.py

```diff
@@ -1070,6 +1078,58 @@ def test_main_view_generates_PM_header(self, mocker, message,
         assert isinstance(view_components[1], Columns)
         assert isinstance(view_components[2], Padding)
 
+    @pytest.mark.parametrize('message, is_empty_narrow', [
+        ({
+            'type': 'stream',
+            'display_recipient': 'foo',
+            'stream_id': 8,
+            'subject': 'disription about foo',
+            'content': "<p> dummy stream message. </p>",
+            'sender_full_name': 'Welcome Bot',
+            'sender_email': 'welcome-bot@zulip.com',
+            'timestamp': 150989984,
+            'id': None,
+            'reactions': [],
+            'flags': ['read']
+        }, True),
+        ({
+            'type': 'private',
+            'display_recipient': [{'full_name': 'Iago',
+                                   'email': 'iago@zulip.com',
+                                   'id': None}],
+            'sender_id': 9,
+            'content': "<p> dummy PM message. </p>",
+            'sender_full_name': 'Welcome Bot',
+            'sender_email': 'welcome-bot@zulip.com',
+            'timestamp': 150999544,
+            'id': None,
+            'reactions': [],
+            'flags': ['read']
+        }, True)
+    ])
+    def test_dummy_message(self, mocker, message,
+                           is_empty_narrow):
```

Comment:
As I said re the other commit, I would expect this test to go in the commit with the feature introduction, not the code refactoring. Generally when refactoring, tests should pass except minor changes - when you add a new feature/fix, then you add more tests to lock that new aspect in place.